src/main/java/org/gaul/s3proxy/S3ProxyHandler.java

```diff
@@ -1490,8 +1490,12 @@ private void handleBlobList(HttpServletRequest request,
                 if (Quirks.OPAQUE_MARKERS.contains(blobStoreType)) {
                     StorageMetadata sm = Streams.findLast(set.stream()).orElse(null);
                     if (sm != null) {
-                        lastKeyToMarker.put(Maps.immutableEntry(containerName,
-                                encodeBlob(encodingType, nextMarker)), nextMarker);
+                        // TODO: verify if we need this handling at all for V2
+                        String lastKey = isListV2 ? encodeBlob(encodingType, nextMarker) : sm.getName();
```

Comment:
@gaul 
As long as `isListV2` works as intended it should be good. Regardless I'd want to test the changes.

Reply:
All good. The change doesn't break pagination.